### PR TITLE
Milestone -2 - add nameable class, base decorator, capitalize and trim decorators

### PR DESCRIPTION
**Pull request by ram1117**
In this PR, I have implemented the following

- create `Nameable` class and define `correct_name` method.
- extend `Person`, `BaseDecorator` classes from `Nameable` class.
- create `CapitalizeDecorator` , `TrimmerDecorator` classes and extend from `BaseDecorator` class.
- define `correct_name` in the above classes with logic defined to meet the project requirement.

### CODE REVIEW

File: modules/person.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
-class Person
+require_relative './nameable'
+class Person < Nameable
```
Comment (Davidosky007) [inline modules/person.rb:2]: * Kindly fix this error I get when I try to test the person decorator, please fix it 
* HINT 💭 you can rewrite your person method like this 
image

image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ram1117 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* proper use of Gitflow 👍 
* Good looking readme 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/person.rb (hunk 1 above)
Comment (ram1117) [reply modules/person.rb:2]: Hi @Davidosky007 . I am using keyword arguments for my constructor. (https://thoughtbot.com/upcase/videos/ruby-keyword-arguments). And for the person object, as per milestone 1 requirement, only 'name' and 'parent_permission' are optional arguments. Age is a required argument. Seems it works fine when I create an object with 'age:' parameter.

image

Comment (Davidosky007) [reply modules/person.rb:2]: Kindly make this work according to the requirement as i get errors trying to test your decoration 
image

Comment (ram1117) [reply modules/person.rb:2]: @Davidosky007 I am running main.rb with the code above. I don't see errors. I have implemented decorator as per project requirement. Also, I am using keyword parameters to initialize the person object.

image

**Comment by medaminedev66** [comment]
Hello @ram1117👋,

_This is an additional review as a result of the second opinion request._

I confirm that your code works as expected and does not show any errors. Great job @ram1117 👏 

- Testing to create a person object:
image

- Running main.rb file:
image


As this was the only reason that your project was not approved it will be approved as soon as possible.
Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approve right away :)

Keep rocking 🚀!

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @ram1117 ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**